caches/clan_test.go

```diff
@@ -0,0 +1,130 @@
+package caches_test
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	uuid "github.com/satori/go.uuid"
+	"github.com/topfreegames/khan/models"
+	"github.com/topfreegames/khan/testing"
+)
+
+var _ = Describe("Clan Cache", func() {
+	var testDb models.DB
+
+	BeforeEach(func() {
+		var err error
+		testDb, err = testing.GetTestDB()
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	Describe("Clans Summaries", func() {
+		It("Should return a cached payload for a second call made immediately after the first", func() {
+			gameID := uuid.NewV4().String()
+			_, clans, err := models.GetTestClans(testDb, gameID, "test-sort-clan", 10)
+			Expect(err).NotTo(HaveOccurred())
+
+			var publicIDs []string
```

Comment:
Suggestion: instead of comparing bits of the result one by one, I'd suggest assembling a list of "wanted" summaries and comparing the response as a whole.

Eg:

```go

import "github.com/kylelemons/godebug/pretty"

want := []map[string]interface{}{
  map[string]interface{}{}, // clan1
  map[string]interface{}{}, // clan2
  // etc
}

got, err := cache.GetClansSummaries(testDb, gameID, publicIDs)
if err != nil {
  t.Fafalf("GetClansSummaries returned err = %v", err)
}
if diff := pretty.Compare(got, want); diff != "" {
  t.Errorf("post-GetClansSummaries diff: (-got +want)\n%v", diff)
}
```

I'm more of a [testing purist](https://github.com/golang/go/wiki/TableDrivenTests) myself, so feel free to adapt the snippet above to the BDD monstrosity we're using here   ;P

Reply:
I'll do it next time!